src/hd.rs

```diff
@@ -181,111 +181,59 @@ impl HDWallet {
         }
     }
 
-    pub fn from_bare_xpub(
```

Comment:
What should I do about these two? Just guess the descriptor based on extended pubkey version (xpub/ypub/zpub etc) and call `from_descriptor`?

Reply:
Basically yes, figure out the appropriate descriptor(s) (two in the case of `--xpub`, one for `--bare-xpub`) and delegate to `from_descriptor`. Note that ypubs/zpubs will have to be provided in their xpub form (this is already case if encode the `ExtendedPubKey` to a string).

There should also be an `ext_key: Option<XyzPubKey>` field on the `Wallet` struct set for wallets that originated from an xpub, for backward compatibility. But lets leave that to later for now.